src/components/UI/Card.js

```diff
@@ -47,110 +47,137 @@ class Card extends React.Component {
       tempData: {
         inputValue,
         textareaValue,
-      }
+      },
     }));
-  }
+  };
 
   onEdit = () => {
     this.setState({
       clicked: true,
       checked: false,
     });
-  }
+  };
 
-  onCheckboxChange = () => {
+  onCheckboxHeaderCardChange = () => {
     this.setState(({ checked }) => ({ checked: !checked }));
+  };
+
+  /* onCheckboxUnderHeaderChange = (props) => {
+    this.setState((state, { inputValue, textareaValue }) => ({
+      clicked: false,
+      tempData: {
+        inputValue,
+        textareaValue,
+      },
+    }));
+  }; */
+
+  static getDerivedStateFromProps(props, state) {
+    if (props.checkboxUnderHeader === true) {
+      return {
+        clicked: false,
+        tempData: { ...props.data },
+      };
+    }
+    return null;
   }
 
   render() {
     const {
       clicked,
       checked,
-      tempData: {
-        inputValue,
-        textareaValue,
-      },
+      tempData: { inputValue, textareaValue },
     } = this.state;
 
-    const inputClass = classnames('card-text__caption-none', {
-      'card-text__caption': clicked,
-    });
-    const btnEditClass = classnames('card-button', {
-      'card-button__none': clicked,
+    const { checkboxUnderHeader } = this.props;
+
+    const inputClass = classnames("card-text__caption-none", {
+      "card-text__caption": clicked,
     });
-    const checkboxClass = classnames({
-      'card-input__none': clicked,
+    const editCheckboxHeaderCardClass = classnames("edit-checkbox__header", {
+      "edit-checkbox__header-none": clicked,
     });
+    /* const checkboxClass = classnames({
+      "card-input__none": clicked,
+    }); */
     const saveCancelButtonsClass = classnames({
-      'save-cancel__buttons': !clicked,
+      "save-cancel__buttons-none": !clicked,
     });
-    const textareaClass = classnames('card-text__none', {
-      'card-text': clicked,
+
+    const textareaClass = classnames("card-text__none", {
+      "card-text": clicked,
     });
```

Comment:
Just rechecked your logic of wirking with classes - missed this last time. You should not use 'display: none' attribute in such a way. The whole React concept is driven by principle to render only those things you need to render. Do not hide elements when you do not need them - just do not render them at all instead.